godotproject/Player.gd

```diff
@@ -400,8 +402,19 @@ func update_anim():
 			facing.x = user_direction.x
 		# At the moment, we only face left or right, no up/down/diagonals.
 		facing.y = 0
+		
 		# If we're facing left, flip the frog.
-		$Sprite.flip_h = facing.x < 0
+		# $Sprite.flip_h = facing.x < 0
+		
+		# Flip based on scale property of root node;
+		# Ensures collision layer flips properly with sprite
+		if velocity.x > 0:
+			scale.x = initial_scale.x * sign(scale.y)
+		elif velocity.x < 0:
+			scale.x = -initial_scale.x * sign(scale.y)
```

Comment:
Quick q, why is scale.y involved here? I can't tell why it would be...
Also I think we can use the facing variable, so maybe the code could look something like:
```suggestion
              scale.x = sign(facing.x) * initial_scale.x
```

Reply:
TBH I'm not 100% sure why scale.y needs to be involved here - could be because the initial scale isn't (1,1)? So far it's the only way I've found that keeps the sprite from flickering between left and right. I tried it out and your suggested change results in this flickering too unless I multiply scale.y. I'm going off of the comments in this issue: https://github.com/godotengine/godot/issues/12335